### ISSUE TITLE
Proposal for a change in the BatchNormOp (Non breaking change)

### ISSUE THREAD

**Issue by owulveryck** [gorgonia/gorgonia#248]
Hello,

_Disclaimer_: I am not entirely aware of the work of the `BatchNorm` and `BatchNormOp` function yet, this needs to be tested in detail.

My understanding is that is that, with the current scope of the `BatchNorm` functions, it's not possible to run a Batchnorm in forward pass with pre-trained learnable (mean, variance for example).

I am proposing a new API that allows the use of the `BatchNormOp` without the need for `BatchNorm` (mainly for the `testing` phase). The changes are:

* Exposing some parameters of the struct such as `Momentum` and `Epsilon` and the learnables
* Creating an init function to initialize the scratch spaces in regard of the input node. (see the [godoc](https://godoc.org/github.com/owulveryck/gorgonia#BatchNormOp.Init)).

This allows me to use a broadcasted `bias` and `mean` on the node given by an `ApplyOp(op, x)` described in the [ONNX spec](https://github.com/onnx/onnx/blob/master/docs/Operators.md#BatchNormalization) (see an example of implementation [here](https://sourcegraph.com/github.com/owulveryck/gorgonnx@2a92c0516c89e6d1db22f437caa3f0509246378d/-/blob/operators/batchnorm_wip.go#L67)).
![batchnorm](https://user-images.githubusercontent.com/11583401/47922987-96ca7700-deb8-11e8-8593-a86752d21de9.png)

The example is not fully tested (there are no tests for the Batchnorm in Gorgonia, and my [generated tests](https://github.com/owulveryck/gorgonnx/blob/9e65d4a3c545df435b7c2f8525ca130be37e694a/operators/batchnorm_autogenerated_test.go) is still red). But before investing more in this design, I'd like your advice to see if I didn't miss anything.

_Note_: I haven't modify the BatchNorm function to use the `Init` function yet.

All the code (including the other PR I have made for the broadcast) are in a branch on my GitHub repo https://github.com/owulveryck/gorgonia.
